lmp/model/_transformer.py

```diff
@@ -0,0 +1,460 @@
+r"""Transformer Language Model with Transformer's encoder architecture."""
+
+
+import math
+import argparse
+from typing import ClassVar, Dict, Optional
+
+import torch
+import torch.nn as nn
+
+from lmp.model._rnn import RNNModel
+from lmp.tknzr._base import BaseTknzr
+
+
+class PositionalEncoding(nn.Module):
+    r"""Positional Encoding.
+
+    In order for the model to make use of the order of the sequence,
+    add ``positional encodings`` to the input embeddings.
+
+    .. math::
+
+            \begin{align}
+            \text{PE}_{(\text{pos},\text{2i})} &= \sin\
+             (\frac{\text{pos}}{10000^\frac{\text{2i}}{d_{\text{model}}}}) \\
+            \text{PE}_{(\text{pos},\text{2i+1})} &= \cos\
+             (\frac{\text{pos}}{10000^\frac{\text{2i}}{d_{\text{model}}}}) \\
+            \end{align}
+
+    Parameters
+    ==========
+    d_hid: int
+        Hidden dimension for MLP and Transformer.
+        Must be bigger than or equal to ``1``.
+    dropout: float
+        Dropout probability.
+        Must satisfy ``0.0 <= dropout <= 1.0``.
+    max_len: int
+        Max length of the input sequence.
+        Must be bigger than or equal to ``1``.
+
+    Attributes
+    ==========
+    dropout: torch.nn.Dropout
+        Positional Encoding dropout.
+    pe: torch.nn.FloatTensor
+        The values of Positional Encoding.
+    """
+
+    def __init__(self, d_hid, dropout, max_len):
```

Comment:
Define function with keywords only parameters and add type annotations.
(change to `def __init__(self, *, d_hid: int, dropout: float, max_seq_len: int):`)